windows/hid.c

```diff
@@ -174,19 +174,42 @@ static int lookup_functions()
 #endif /* HIDAPI_USE_DDK */
 
 struct hid_device_ {
-		HANDLE device_handle;
-		BOOL blocking;
-		USHORT output_report_length;
-		unsigned char *write_buf;
-		size_t input_report_length;
-		USHORT feature_report_length;
-		unsigned char *feature_buf;
-		wchar_t *last_error_str;
-		BOOL read_pending;
-		char *read_buf;
-		OVERLAPPED ol;
-		OVERLAPPED write_ol;
-		struct hid_device_info* device_info;
+	HANDLE device_handle;
+	BOOL blocking;
+	USHORT output_report_length;
+	unsigned char *write_buf;
+	size_t input_report_length;
+	USHORT feature_report_length;
+	unsigned char *feature_buf;
+	wchar_t *last_error_str;
+	BOOL read_pending;
+	char *read_buf;
+	OVERLAPPED ol;
+	OVERLAPPED write_ol;
+	struct hid_device_info* device_info;
+};
+
+static struct hid_hotplug_context {
+	/* Win32 notification handle */
+	HCMNOTIFICATION notify_handle;
+
+	/* Win32 mutex handle, for both cached device list and callback list changes */
+	HANDLE mutex;
+
+	/* HIDAPI unique callback handle counter */
+	hid_hotplug_callback_handle next_handle;
+
+	/* Linked list of the hotplug callbacks */
+	struct hid_hotplug_callback *hotplug_cbs;
+
+	/* Linked list of the device infos (mandatory when the device is disconnected) */
+	struct hid_device_info *devs;
+} hid_hotplug_context = {
+	.notify_handle = NULL,
+	.mutex = NULL,
+	.next_handle = 1,
```

Comment:
what is `1`? can we use a named constant instead? `INVALID_HANDLE_VALUE` perhaps?